drivers/video/hm01b0.c

```diff
@@ -239,32 +415,231 @@ static int hm01b0_soft_reset(const struct device *dev)
 	return ret;
 }
 
+static int hm01b0_set_frmival(const struct device *dev, struct video_frmival *frmival)
+{
+	const struct hm01b0_config *config = dev->config;
+	struct hm01b0_data *drv_data = dev->data;
+	int ret;
+	uint32_t osc_div = 0;
+	bool highres = false;
+	struct video_frmival_enum fie;
+
+	if (drv_data->fmt.width == 320 && drv_data->fmt.height == 320) {
+		highres = true;
+	}
+
+	fie.format = &drv_data->fmt;
+	fie.discrete = *frmival;
+	video_closest_frmival(dev, &fie);
+
+	/*
+	 * Note: in highres mode max FPS is 45 else 60
+	 *  OSC clock divider: 00=/8, 01=/4, 10=/2, 11=/1
+	 */
+
+	switch (fie.index) {
+	case HM01B0_60_FPS:
+		osc_div = 0x03;
+		break;
+	case HM01B0_30_FPS:
+		osc_div = highres ? 0x03 : 0x02;
+		break;
+	case HM01B0_15_FPS:
+		osc_div = highres ? 0x02 : 0x01;
+		break;
+	case HM01B0_8_FPS:
+		osc_div = 0x00;
+		break;
+	default:
+		CODE_UNREACHABLE;
+	}
+
+	/* We also set the VT_REG_DIV bits[3:2] to 10 (0x2) or /1 */
+	osc_div |= HM01B0_CCI_OSC_CLOCK_DIV_VT_REG_DIV_1;
+
+	ret = video_write_cci_reg(&config->i2c, HM01B0_CCI_OSC_CLOCK_DIV, osc_div);
+	if (ret < 0) {
+		LOG_ERR("Failed to write OSC_CLK_DIV = %x reg (%d)", osc_div, ret);
+		return ret;
+	}
+
+	/* GRP_PARAM_HOLD */
+	ret = video_write_cci_reg(&config->i2c, HM01B0_CCI_GRP_PARAM_HOLD, 0x01);
+	if (ret < 0) {
+		LOG_ERR("Failed to write GRP_PARAM_HOLD reg (%d)", ret);
+		return ret;
+	}
+
+	drv_data->cur_frmrate = frmival->numerator;
+
+	LOG_DBG("FrameRate selected: %d", frmival->numerator);
+	LOG_DBG("HIRES Selected: %d", highres);
+	LOG_DBG("OSC DIV: %d", osc_div);
+
+	return 0;
+}
+
+static int hm01b0_get_frmival(const struct device *dev, struct video_frmival *frmival)
+{
+	struct hm01b0_data *drv_data = dev->data;
+
+	frmival->numerator = drv_data->cur_frmrate;
+	frmival->denominator = 1;
+
+	return 0;
+}
+
+int hm01b0_enum_frmival(const struct device *dev, struct video_frmival_enum *fie)
+{
+	switch (fie->index) {
+	case HM01B0_60_FPS:
+		fie->discrete.denominator = 60;
+		break;
+	case HM01B0_30_FPS:
+		fie->discrete.denominator = 30;
+		break;
+	case HM01B0_15_FPS:
+		fie->discrete.denominator = 15;
+		break;
+	case HM01B0_8_FPS:
+		fie->discrete.denominator = 8;
+		break;
+	default:
+		return -EINVAL;
+	}
```

Comment:
As seen here https://github.com/zephyrproject-rtos/zephyr/pull/98154/files#r2500836963 the frame rate depends on the resolution

This is the reason why `video_frmival_enum` contains a `fie->fmt` field, so that the framerate returned can be made specific to the format.

For instance:

```suggestion
	bool highres = (drv_data->fmt.width >= 320 && drv_data->fmt.height >= 320);

	switch (fie->index) {
	case HM01B0_45_OR_60_FPS:
		fie->discrete.denominator = highres ? 45 : 60;
		break;
	case HM01B0_22_OR_30_FPS:
		fie->discrete.denominator = highres ? 22 : 30;
		break;
	case HM01B0_11_OR_15_FPS:
		fie->discrete.denominator = highres ? 11 : 15;
		break;
	case HM01B0_5_OR_8_FPS:
		fie->discrete.denominator = highres ? 5 : 8;
		break;
	default:
		return -EINVAL;
	}
```

Reply:
@KurtE 
Was testing this out on the 360 and found that at least set_frmival was correct in terms of configuring the framerate reg values.  

However, when I was testing on the GIga through the arduino ide I was not seeing much change in fps. 
For 10 fps seeing about 26 fps
For 15 seeing about 25 fps
For 30 seeing about 26 fps.
At 60 seeing about 12 fps which is strange to me

So looks like no matter what something else is off